pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2921,6 +2921,28 @@ type CephFilesystemSubVolumeGroupSpec struct {
 	// list of Ceph Filesystem volumes with `ceph fs volume ls`. To learn more about Ceph Filesystem
 	// abstractions see https://docs.ceph.com/en/latest/cephfs/fs-volumes/#fs-volumes-and-subvolumes
 	FilesystemName string `json:"filesystemName"`
+	// Pinning configuration of CephFilesystemSubVolumeGroup,
+	// reference https://docs.ceph.com/en/quincy/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups
+	// +optional
+	Pinning CephFilesystemSubVolumeGroupSpecPinning `json:"pinning,omitempty"`
+}
+
+// CephFilesystemSubVolumeGroupSpecPinning represents the pinning configuration of SubVolumeGroup
+// +kubebuilder:validation:XValidation:message="only one pinning type should be set",rule="(has(self.export) && !has(self.distributed) && !has(self.random)) || (!has(self.export) && has(self.distributed) && !has(self.random)) || (!has(self.export) && !has(self.distributed) && has(self.random)) || (!has(self.export) && !has(self.distributed) && !has(self.random))"
+type CephFilesystemSubVolumeGroupSpecPinning struct {
+	// +kubebuilder:validation:Minimum=-1
+	// +kubebuilder:validation:Maximum=256
+	// +optional
+	// +nullable
+	Export *int `json:"export,omitempty"`
+	// +optional
+	// +nullable
+	Distributed *bool `json:"distributed,omitempty"`
```

Comment:
Since the other values are numerical, let's keep this as an int and then restrict to 0 or 1. It seems inconsistent if this is a bool.
```suggestion
	Distributed *int `json:"distributed,omitempty"`
```